pkg/controllers/disruption/consolidation.go

```diff
@@ -246,9 +258,16 @@ func (c *consolidation) computeSpotToSpotConsolidation(ctx context.Context, cand
 	}
 
 	// For single-node consolidation:
-	// We check whether we have 15 cheaper instances than the current candidate instance. If this is the case, we know the following things:
-	//   1) The current candidate is not in the set of the 15 cheapest instance types and
-	//   2) There were at least 15 options cheaper than the current candidate.
+
+	// If a user has minValues set in their NodePool that require less than 15 instance types, then the default 15 instance type minimum will continue to be enforced to enable a spot-to-spot consolidation.
+	// If minValues for the number of instance types is beyond 15, then number of instance types from minValues would be maintained to enable spot-to-spot consolidation.
+	if results.NewNodeClaims[0].Requirements.Has(v1.LabelInstanceTypeStable) && results.NewNodeClaims[0].Requirements.Get(v1.LabelInstanceTypeStable).MinValues != nil {
```

Comment:
What's the need for this check? Shouldn't the check that we have at least 15 instance types be enough for this to succeed on its own?

Reply:
I'm assuming that you added this to support truncating, but given the fact that no one is required to set the instance type requirement directly, I think we just have to throw-out the truncation of the instance types that we use for launch during consolidation. I think we simply have to accept that we will most likely get the bottom 15 instance types when we make a consolidation call and, if we don't, we'll just walk down a bit until we get to the bottom 15